### PR TITLE
feed v2: improve and extend for meta data documentation

### PR DESCRIPTION
**Pull request by mswart**
Extend the feed for meta data as discussed in openmensa/openmensa#4.

Open for comments. Anything unclear or unclear expressed. Feel free to mark/correct typos and grammar errors.

I have the two open questions:
- [x] Which telephone format do we want to recommend? If you prefer a every + very common one (`0331 977-3749`)
- [x] how is the coordinate system named we need coordinates for
- [x] is the naming `public` / `restricted` better?

### CODE REVIEW

File: content/feed/v2.md [hunk 1]
```diff
@@ -18,19 +18,60 @@ If you build your feed's xml carefully (simply copy the first 5 lines from our e
...
 
-## Changes to [v1](/feed/v1/)
+## *openmensa* root element
 
-* new *openmensa* root element
-* *name*s need to have less than 250 chars
-* rename cafeteria to *canteen*
-* price *role*s
-* closing times on day granularity
+The root element is `openmensa`. It encapsulate all other attributes. This element will be preserved across future feed versions. The *version* attribute must equal 2.0.
```
Comment (jgraichen) [inline content/feed/v2.md]: As the version can be alternativly specified by a HTTP header I'd drop the "attribute" here to clarify that only the version must equal "2.0.": `The *version* must equal 2.0.`

**Comment by jgraichen** [comment]
**public/restricted**: I would like to not have two tags for `public` or `restricted` but only `<public>true</public>` or `<public>false</public>`. Or `<availability>public</availability>`. Imho "public" and "restricted" should be values not keys.

**telephone**: If we handle it as a free text field we could simply recommend [E.123](https://en.wikipedia.org/wiki/E.123) and DIN 5008.

**Comment by jgraichen** [comment]
@cmur2 coordinate systems?

**Comment by cmur2** [comment]
@mswart in this step you also want to purge the old "changes to v1"? do you know the situations of the parsers? all on v2 now? do we require v2?

**Comment by mswart** [comment]
My main reason is: new readers / developers understand / read the v2 feed description without knowing the feed v1.

I did not do a real analysis, but I expect that most of the parsers are v2. We started with v2 while I was developing my parsers and most of our current parsers were developed after that. And at least Kai migrated successful from v1 to v2.

I will keep v1 support in openmensa unless it is much pain but we should start deprecating it more directly - e.g. do we really want two have API v1 and Feed v1 directly linked from the top bar? @jgraichen?

**Comment by kaifabian** [comment]
I'd also vote for further deprecation of v1 in the docs. Since what we're specifying now is basically a "Feed v2.5" (even though we do not *have* to name it like this, since all proposed changes will be fully compatible to Feed v2, we should think about that to give devs a chance to see what changed, shoud we not?), IMHO v1 should be moved to a separate button (like: "obsolete versions").

We should also try to check all feeds currently listed in OM on what version they provide. If we have no v1 feeds left, I'd even vote for a complete drop of v1 support - the sooner, the better.

**Comment by cmur2** [comment]
After thinking about it I second @kaifabian - the changes from v2 to "v2.5" should be visible to allow those who already known v2 to skip to the interessing parts. Hiding the docs for v1 is desireable but I don't want them to be removed completely, for historical reasons. An assessment about which v1 feeds are currently in the system would be nice.

**Comment by mswart** [comment]
First any reason why not to choose 2.1 as version?

I propose add a short v2 specific change log (e.g. 2012-08-x released 2.0, 2015-04-x released 2.1: new version tag + meta data) and add `(since 2.1)` to version and meta data headline.

My concern is whether these more precise version number are confusion? We still have only one xml schema for all 2.x versions, or? What should be the value of the openmensa version attribute? Always `2.0` or `2.0`, `2.1` (`2.x`) depending on which new feature the parser implemented? Do we raise an error on a to high value (e.g. currently on 2.2). 

While writing this comment I prefer support those specific versions (2.0, 2.1) and raise an error on two high versions.

**Comment by cmur2** [comment]
I saw "v2.5" as a working title as long as we didn't agree on a specific (minor) version. Keeping the same XML schema for a major version seems plausible. Therefore too high version should be an error. The version attribute should state 2.0 or 2.1 etc.
The changelog is a good idea.

**Comment by mswart** [comment]
@cmur2 @kaifabian I have updated the PR and added changelog + version hints

**Comment by mswart** [comment]
@jgraichen @kaifabian I added a specification for opening times. Please take a look. 

@jgraichen I tried to use ISO-8601 to specify the interval, but I was not able to find documentation/support how to correctly specify time intervals. So I assume that other people will also have trouble with it - I think using a standard would be nice but creates more issues than it solves.

**Pull request merged**